WorkingwithAnsibleAdHocCommands.md

```diff
@@ -0,0 +1,210 @@
+
+This is a good place to start to understand the basics of what Ansible can do prior to learning the playbooks language. Ad-hoc commands can also be used to do quick things that you might not necessarily want to write a full playbook for.
+
+This guide will be focused mainly on the application of these commands on Network devices, and in our case its for EXOS devices.
+
+## Pre-requisite:
+
+A working setup of Ansible system with at least one EXOS device.
+
+> Refer [this](https://github.com/extremenetworks/ansible-extreme/blob/master/GettingStartedwithAnsibleusingGNS3.md) guide to prepare a similar setup on GNS3
+
+## Basic Ansible Commands
+
+### Check Ansible Version
+
+```ansible --version```
+
+Run this command to check the version of ansible currently installed on a system. This also shows other details such as python version, ansible configuration file path etc.
+
+Sample command and output:
+
+```sh
+(venv) root@Net:~# ansible --version
+ansible 2.7.0.dev0 (devel 0102e42272) last updated 2018/06/17 04:22:52 (GMT +000)
+  config file = /root/ansible.cfg
+  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
+  ansible python module location = /root/ansible/lib/ansible
+  executable location = /root/ansible/bin/ansible
+  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
+```  
+### Check available hosts in Ansible Inventory
+
+```ansible --list-hosts all```
+
+Run this command to check all the available hosts in Ansible's Inventory. In order to check the hosts available in a particular group, replace the word ``'all'`` with the name of the group.
+
+Sample command and output:
+
+```sh
+(venv) root@Net:~# ansible --list-hosts all
+  hosts (5):
+    S1
+    S2
+    S3
+    S4
+    S5
+```
+### Ansible Documentation
+
+```ansible-doc <module_name>```
+
+This command shows the documentation of a particular module. This includes, supported arguments, few examples and return values of the module. For example, run below command to get the documentation for "exos_command" module
+
+```ansible-doc exos_command```
+
+### Ansible Help
+
+```ansible -h```
+
+Run this command to see all the available options under ```ansible```.
+
+Refer [this](https://docs.ansible.com/ansible/latest/user_guide/command_line_tools.html) document for more details on working with Command Line Tools
+
+## Basic Ansible Ad-Hoc Commands
+
+### Pinging Hosts
+
+```ansible S1 -m ping```
+
+Use this command to check if the hosts are reachable from ansible.
+
+Sample command and output below:
+
+```sh
+(venv) root@Net:~# ansible S1 -m ping
+S1 | SUCCESS => {
+    "changed": false,
+    "ping": "pong"
+}
+```
+
+### Gathering Facts
+
+```ansible <device> -m setup```
+
+Use this command to gather different facts of the device. Replace ```device``` with the group name or hostname of the device for which the detail needs to be gathered.
+
+Sample command and **truncated** output below:
+
```

Comment:
I'm a bit surprised this command works. I thought it would only work for remote servers. Learn something new every day I guess.

Reply:
I still have a few questions around this. I am checking on that on Ansible Forums.